File: app/src/main/java/io/github/shadow578/music_dl/ui/splash/SplashScreenActivity.java
package io.github.shadow578.music_dl.ui.splash;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import io.github.shadow578.music_dl.ui.main.MainActivity;
import io.github.shadow578.music_dl.util.Async;

/**
 * basic splash- screen activity.
 * displays a splash- screen, then redirects the user to the correct activity
 */
public class SplashScreenActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // minimum time of 1 seconds
        Async.runLaterOnMain(() -> {
            startActivity(new Intent(this, MainActivity.class));
            finish();
        }, 1000);
    }
}

File: app/src/main/kotlin/io/github/shadow578/yodel/ui/splash/SplashScreenActivity.kt
package io.github.shadow578.yodel.ui.splash

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import io.github.shadow578.yodel.ui.main.MainActivity
import io.github.shadow578.yodel.util.launchMain
import kotlinx.coroutines.delay

/**
 * basic splash- screen activity.
 * displays a splash- screen, then redirects the user to the correct activity
 */
class SplashScreenActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        launchMain {
            delay(50)
            startActivity(Intent(this@SplashScreenActivity, MainActivity::class.java))
            finish()
        }
    }
}